packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -253,7 +253,7 @@ Future<GradleProject> _readGradleProject(
       environment: gradleEnv,
     );
     project = GradleProject.fromAppProperties(propertiesRunResult.stdout, tasksRunResult.stdout);
-  } catch (exception) {
+  } on Exception catch (exception) {
```

Comment:
It looks like this should only be catching `ProcessException`. We'd want a crash report sent to let us know that we are failing to parse `propertiesRunResult.stdout`. I suspect that is what leads to the `StateError` here, but the problem is obscured by the `TypeError`.